web-common/src/features/entity-management/file-artifacts-store.ts

```diff
@@ -1,131 +1,297 @@
+      this.getAllErrors(queryClient, instanceId),
+      (errors) => errors.length > 0,
     );
-  },
+  }
+
+  public getResourceStatusStore(
+    queryClient: QueryClient,
+    instanceId: string,
+    validator?: (res: V1Resource) => boolean,
+  ) {
+    return derived(
+      [
+        this.getResource(queryClient, instanceId),
+        this.getAllErrors(queryClient, instanceId),
+        useProjectParser(queryClient, instanceId),
+      ],
+      ([resourceResp, errors, projectParserResp]) => {
+        if (projectParserResp.isError) {
+          return {
+            status: ResourceStatus.Errored,
+            error: projectParserResp.error,
+          };
+        }
+
+        if (
+          errors.length ||
+          (resourceResp.isError && !resourceResp.isFetching) ||
+          projectParserResp.isError
+        ) {
+          return {
+            status: ResourceStatus.Errored,
+            error: resourceResp.error ?? projectParserResp.error,
+          };
+        }
 
-  setIsReconciling(affectedPath: string, isReconciling: boolean) {
-    createOrUpdateFileArtifact(
-      affectedPath,
-      (entityData: FileArtifactsData) => {
-        entityData.isReconciling = isReconciling;
+        let isBusy: boolean;
+        if (validator && resourceResp.data) {
+          isBusy = !validator(resourceResp.data);
+        } else {
+          isBusy =
+            resourceResp.isFetching ||
+            resourceResp.data?.meta?.reconcileStatus !==
+              V1ReconcileStatus.RECONCILE_STATUS_IDLE;
+        }
+
+        return {
+          status: isBusy ? ResourceStatus.Busy : ResourceStatus.Idle,
+          resource: resourceResp.data,
+        };
       },
     );
-  },
-};
-
-export type FileArtifactsStore = Readable<FileArtifactsState> &
-  typeof fileArtifactsEntitiesReducers;
-
-export const fileArtifactsStore: FileArtifactsStore = {
-  subscribe,
-  ...fileArtifactsEntitiesReducers,
-};
-
-export function getIsFileReconcilingStore(file: string) {
-  return derived(
-    fileArtifactsStore,
-    ($store) => $store.entities[file]?.isReconciling,
-  );
-}
+  }
 
-function correctFilePath(filePath: string) {
-  // TODO: why does affectedPaths not have the leading /
-  if (!filePath.startsWith("/")) {
-    filePath = "/" + filePath;
+  private updateNameIfChanged(resource: V1Resource) {
+    const curName = get(this.name);
+    if (
+      curName?.name !== resource.meta?.name?.name ||
+      curName?.kind !== resource.meta?.name?.kind
+    ) {
+      this.name.set({
+        kind: resource.meta?.name?.kind,
+        name: resource.meta?.name?.name,
+      });
+    }
   }
-  return filePath;
 }
 
-export function getFileArtifactReconciliationErrors(
-  fileState: FileArtifactsState,
-  fileName: string,
-): V1ReconcileError[] {
-  const path = Object.keys(fileState?.entities)?.find((key) => {
-    return key.endsWith(fileName);
-  });
-  return fileState?.entities?.[path]?.errors;
+export class FileArtifactsStore {
```

Comment:
Nit: this isn't a store, so can we name it just `FileArtifacts`? And maybe change the file name too to `entity-management/file-artifacts.ts`?